src/widgets/mod.rs

```diff
@@ -30,6 +30,7 @@ mod reflow;
 mod sparkline;
 mod table;
 mod tabs;
+pub mod tree;
```

Comment:
The Tree* types are re-exported below:
```rust
pub use self::tree::{Tree, TreeIdentifier, TreeItem, TreeState};
```
So these show up in the widgets as `ratatui::widgets::{Tree, TreeIdentifier, TreeItem, TreeState}` which is consitent with the way that we expose the rest of the widgets in the library.

I wasn't sure that exposing `tree::flatten::{flatten, Flattened}` and `tree::identifier::{get_without_leaf as get_identifier_without_leaf, TreeIdentifierVec}` without them being prefixed with `Tree` was a good idea (though I missed the `TreeIdentifierVec` there - which could be exported fine). My rationale is what does it look like having the struct / method in a tab completion list. Happy to change my mind on this if you think otherwise.

Reply:
I think when searching for tree thingies it's also quite easy to find `ratatui::widgets::tree::` via tab completion. Personally I would avoid exposing the same things on multiple places as it might end up with different code for the same thing.

Personally I would just go with exposing the tree module and don't reexport them again. Then nothing can be missed accidentally for example and its a bit more grouped rather than exposing a bunch of tree related stuff on the widget module.

Reply:
> Personally I would just go with exposing the tree module and don't reexport them again. Then nothing can be missed accidentally for example and its a bit more grouped rather than exposing a bunch of tree related stuff on the widget module.

I agree with your logic on this.